File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string original = "";
        for (int i = 0; i < s.length(); ++i) {
            if (isalnum(s[i])) {
                original += tolower(s[i]);
            }
        }
        string reversed = original;
        reverse(reversed.begin(), reversed.end());

        return original == reversed;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder original = new StringBuilder();
        for (int i = 0; i < s.length(); ++i) {
            char ch = s.charAt(i);
            if ((ch >= 'a' && ch <= 'z')  || (ch >= '0' && ch <= '9')) {
                original.append(ch);
            } else if (ch >= 'A' && ch <= 'Z') {
                original.append((char)(ch + 32));
            }
        }
        return original.toString().equals(original.reverse().toString());
    }
}